design/authn-ldap.md

```diff
@@ -0,0 +1,236 @@
+    - [Test Cases](#test-cases)
+      - [Unit Tests](#unit-tests)
+      - [E2E tests](#e2e-tests)
+      - [Security testing](#security-testing)
+      - [Performance testing](#performance-testing)
+  - [Logs](#logs)
+  - [Documentation](#documentation)
+  - [Version update](#version-update)
+  - [Security](#security)
+  - [Audit](#audit)
+  - [Solution Review](#solution-review)
+
+## Useful Links
+
+| Link | Private |
+|------|:-------:|
+| [Conjur Docs: LDAP Authentication - Overview](https://docs.conjur.org/Latest/en/Content/Integrations/ldap/ldap-sync-authn-integration.htm) | No |
+| [Conjur Docs: Configure Developer Environment to Use LDAP Authentication](https://docs.conjur.org/Latest/en/Content/Integrations/ldap/ldap-authn-configure-dev-env.htm) (For Ruby-based) | No |
+| [Pseudo-code for AuthenticationStrategy interface](https://gist.github.com/jvanderhoof/c0457452ec83dbf5c31527edfb8c1dcb) | No |
+
+## Overview
+
+Conjur supports various forms of authentication, including LDAP. This document describes the design and implementation of the LDAP authentication
+support in the new Pytyon-based Conjur CLI.
+
+### Support in previous CLI versions
+
+In the Ruby-based CLI, it was possible to configure the CLI to use LDAP authentication instead of the default authentication method. This was
+done by changing the authn_url configured for the CLI, either in .conjurrc (`authn_url: https://<host>/authn-ldap/<serviceid>`) or by setting the
+`CONJUR_AUTHN_URL` environment variable. This worked because the authn-ldap service has the same interface as the default authn service, with the
+exception of the service_id which is included in the URL. Therefore it didn't matter to the CLI which method was being used.
+
+However, with the new Python-based CLI, the authn_url is no longer directly configurable. Instead, we store only the Conjur base url (conjur_url)
+in .conjurrc and construct the authn_url from that in code. To maintain this paradigm, we need to explicitly specify that we want to use LDAP
+when calling the CLI, and have the CLI add the authn-ldap endpoing and service_id to the authn_url.
+
+### Architectural motivation
+
+Another motivation for this change is that we want to add support for all forms of authentication in the various Conjur SDKs. Therefore we want
+the conjur-api-python library to be constructed in a way that allows us to easily add support for different authentication methods by
+implementing a common `AuthenticationStrategy` interface. Since the CLI and API are now in separate repositories, much of the implementation for
+this design will occur in the API, which will move us closer to a pluggable authentication flow.
+
+## Solution
+
+### User Experience
+
+The `conjur login` command should support LDAP authentication. It should be possible to specify the authentication type using the
+`-t` or `--type` option. The default value should be `authn` (same as current), and it should additionally support `ldap`.
```

Comment:
I like `-t` as the short flag option. I also like the descriptiveness of the `--authn-type` flag suggested below.